Document the `http_headers_getsetcookie` and `urlsearchparams_delete_… (#9235)
cloudflare/cloudflare-docs: content/workers/_partials/_platform-compatibility-dates/http_headers_getsetcookie.md

---
_build:
  publishResources: false
  render: never
  list: never

name: "Header class getSetCookie() method"
sort_date: "2023-03-01"
enable_date: "2023-03-01"
experimental: false
enable_flag: "http_headers_getsetcookie"
disable_flag: "no_http_headers_getsetcookie"
---

The WHATWG introduced the new `getSetCookie()` method on the `Headers` class. Because the method is
new and changes the behavior of the `Headers` class key/value iterators there is a risk of breaking
existing code. To mitigate this risk, the new behavior will be opt-in via a compatibility flag

> that became the default on March 1, 2023.

Author: ```suggestion
that became the default on March 1, 2023.

Without this flag, the iterator on the `Headers` object will improperly combine the values of
`Set-Cookie` headers into a single item. With the flag enabled, the iterator appropriately keeps
`Set-Cookie` headers separate while continuing to combine other headers.
```

---
_build:
  publishResources: false
  render: never
  list: never

name: "Header class getSetCookie() method"
sort_date: "2023-03-01"
enable_date: "2023-03-01"
experimental: false
enable_flag: "http_headers_getsetcookie"
disable_flag: "no_http_headers_getsetcookie"
---

> The WHATWG introduced the new `getSetCookie()` method on the `Headers` class. Because the method is

Reviewer: Maybe link to the method definition in WHATWG spec so that people can read up on it.

```suggestion
The WHATWG introduced the new [`getSetCookie()` method](https://fetch.spec.whatwg.org/#dom-headers-getsetcookie) on the `Headers` class. Because the method is
```

new and changes the behavior of the `Headers` class key/value iterators there is a risk of breaking

> existing code. To mitigate this risk, the new behavior will be opt-in via a compatibility flag

Reviewer: ```suggestion
existing code. To mitigate this risk, the new behavior is opt-in via the `http_headers_getsetcookie` compatibility flag.
```

> that became the default on March 1, 2023.

Reviewer: ```suggestion
If your compatibility date in is set to March 1, 2023 or after, this new behavior is on by default.
```
